### DISCUSSION TITLE
Latest Docker image: Container ID cannot be mapped to a host ID

### DISCUSSION THREAD

**Discussion by nslusher-sf** [renovatebot/renovate#20854] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

The latest docker image

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

Runs started failing this morning which seems to be because of a new release. We use the daniel-shuy/renovate@3.1.0 CircleCI orb to run Renovate. It is downloading the latest Docker image.

See the log output.

### Relevant debug logs

<details><summary>Logs</summary>

```
Build-agent version 1.0.161799-342e6896 (2023-03-07T15:48:58+0000)
System information:
 Server Version: 20.10.18
 Storage Driver: overlay2
  Backing Filesystem: xfs
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Kernel Version: 5.15.0-1030-aws
 Operating System: Ubuntu 20.04.5 LTS
 OSType: linux
 Architecture: x86_64

Starting container renovate/renovate:latest
Warning: No authentication provided, using CircleCI credentials for pulls from Docker Hub.
  image cache not found on this host, downloading renovate/renovate:latest
latest: Pulling from renovate/renovate

64472391: Already exists 
e5e65d45: Pulling fs layer 
4ba4936a: Pulling fs layer 
7b418c46: Pulling fs layer 
5e356d86: Pulling fs layer 
e0eb92ca: Pulling fs layer 
711c5c27: Pulling fs layer 
e281dba2: Pulling fs layer 
aaec8668: Pulling fs layer 
f621c072: Pulling fs layer 
e0eb92ca: Waiting fs layer 
522b6e70: Pulling fs layer 
711c5c27: Waiting fs layer 
e356d86: Waiting fs layer 
e281dba2: Waiting fs layer 
2029fd8d: Waiting fs layer 
eb1c3afc: Pulling fs layer 
522b6e70: Waiting fs layer 
43679522: Pulling fs layer 
034dcd4b: Waiting fs layer 
eb1c3afc: Waiting fs layer 
11f03cc2: Pulling fs layer 
8b5bfefe: Pulling fs layer 
43679522: Waiting fs layer 
24a1ebc6: Waiting fs layer 
8677fc4d: Waiting fs layer 
8b5bfefe: Waiting fs layer 
1978b6ee: Waiting fs layer 
3e03c1c1: Waiting fs layer 
fb11a688: Pulling fs layer 
bded8fe7: Pulling fs layer 
43b98952: Pulling fs layer 
bded8fe7: Waiting fs layer 
cfabcefb: Download complete ============================================>]  27.78MB/27.78MBB

CircleCI was unable to start the container because of a userns remapping failure in Docker.

This typically means the image contains files with UID/GID values that are higher than Docker and CircleCI can use.

Checkout our docs https://circleci.com/docs/2.0/high-uid-error/ to learn how to fix this problem.
Original error: CircleCI was unable to start the container because of a userns remapping failure in Docker.

This typically means the image contains files with UID/GID values that are higher than Docker and CircleCI can use.

Checkout our docs https://circleci.com/docs/2.0/high-uid-error/ to learn how to fix this problem.
Original error: failed to register layer: Error processing tar file(exit status 1): Container ID 2119470584 cannot be mapped to a host ID
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by HonkingGoose** [comment]
Does the `daniel-shuy/renovate@3.1.0` Orb support Renovate `v35`? The orb was last updated on _Oct 31, 2021_, so maybe it needs to be updated to account for our breaking changes?

**Comment by nslusher-sf** [comment]
I believe it supports it - it is a generic runner that downloads the latest Docker image by default. The documentation that CircleCI lists [above](https://circleci.com/docs/2.0/high-uid-error/) suggests it is something that needs to be fixed by the image maintainers.

**Comment by HonkingGoose** [comment]
Oh okay, just wanted to get the dumb stuff out of the way first. 😄 

I think @viceice knows more about our Docker images, so I'll let them respond to this issue. 😉

**Comment by viceice** [comment]
it's not a renovate issue, it's a circle ci issue. nothing docker related changed

**Reply by viceice** [reply]
please ask the maintainer of the orb for help.

**Reply by nslusher-sf** [reply]
This isn't an orb issue. There are files in the latest Docker image that have a UID/GID outside of the allowed range. I opened up the latest image and ran the commands that CircleCI listed above to identify the problem files:

```
$ find / \( -uid 2119470584 \) -ls 2>/dev/null
  3049471     16 -rw-r--r--   1 2119470584 2042662593    14895 Mar 10 13:00 /opt/buildpack/tools/lerna/6.5.1/node_modules/external-editor/node_modules/tmp/lib/tmp.js
  3049472      4 -rw-r--r--   1 2119470584 2042662593      751 Mar 10 13:00 /opt/buildpack/tools/lerna/6.5.1/node_modules/external-editor/node_modules/tmp/package.json
  3049468      4 -rw-r--r--   1 2119470584 2042662593     1082 Mar 10 13:00 /opt/buildpack/tools/lerna/6.5.1/node_modules/external-editor/node_modules/tmp/LICENSE
  3049469     12 -rw-r--r--   1 2119470584 2042662593     9293 Mar 10 13:00 /opt/buildpack/tools/lerna/6.5.1/node_modules/external-editor/node_modules/tmp/README.md
```

Setting ownership of the affected files to root:root in a new image should fix this issue. You can see that the lib directory is owned by root but the affected files are not:
```
$ pwd
/opt/buildpack/tools/lerna/6.5.1/node_modules/external-editor/node_modules/tmp
$ ls -lah
total 32K
drwxr-xr-x 3 root       root       4.0K Mar 10 13:00 .
drwxr-xr-x 3 root       root       4.0K Mar 10 13:00 ..
-rw-r--r-- 1 2119470584 2042662593 1.1K Mar 10 13:00 LICENSE
-rw-r--r-- 1 2119470584 2042662593 9.1K Mar 10 13:00 README.md
drwxr-xr-x 2 root       root       4.0K Mar 10 13:00 lib
-rw-r--r-- 1 2119470584 2042662593  751 Mar 10 13:00 package.json
```

**Reply by viceice** [reply]
as commented below, it's a npm bug, not a renovate bug. 🤷‍♂️

**Reply by viceice** [reply]
https://github.com/mend/renovate-on-prem/issues/258#issuecomment-1441990685

**Reply by nslusher-sf** [reply]
True, but would it be possible to add a workaround in CI/CD to get the latest Docker image to work correctly?

**Reply by HonkingGoose** [reply]
I found this:

> If anyone else is having this issue. Just be aware of this release of pacote not yet being in npm despite it being released.
> 
>     * npm bundles it's pacote dependency in `node_modules/`
> 
>     * That bundle in npm 9.5.0 doesn't include this fix.
> 
> 
> You can force the update with `npm install -g pacote@latest`

Source:

- https://github.com/npm/pacote/pull/261#issuecomment-1440289679

**Comment by viceice** [comment]
it's lerna again 😕 
```sh
> docker run --rm -it renovate/renovate bash                                                                                   
ubuntu@6c99d4c887d1:/usr/src/app$ find / \( -uid 2119470584 \)  -ls 2>/dev/null
   197684      4 -rw-r--r--   1 2119470584 2042662593      751 Mar 10 13:00 /opt/buildpack/tools/lerna/6.5.1/node_modules/external-editor/node_modules/tmp/package.json
   197680      4 -rw-r--r--   1 2119470584 2042662593     1082 Mar 10 13:00 /opt/buildpack/tools/lerna/6.5.1/node_modules/external-editor/node_modules/tmp/LICENSE
   197681     12 -rw-r--r--   1 2119470584 2042662593     9293 Mar 10 13:00 /opt/buildpack/tools/lerna/6.5.1/node_modules/external-editor/node_modules/tmp/README.md
   197683     16 -rw-r--r--   1 2119470584 2042662593    14895 Mar 10 13:00 /opt/buildpack/tools/lerna/6.5.1/node_modules/external-editor/node_modules/tmp/lib/tmp.js
```

**Reply by viceice** [reply]
- https://github.com/npm/cli/issues/6173

**Comment by viceice** [comment, marked as answer]
- https://github.com/renovatebot/docker-renovate-full/pull/1489

that should workaround the issue

**Reply by nslusher-sf** [reply]
It's working! Thanks a lot for the quick turnaround!
